src/org/opendatakit/briefcase/ui/PullTransferPanel.java

```diff
@@ -293,7 +295,7 @@ public void focusLost(FocusEvent e) {
         .addGroup(
             groupLayout
                 .createParallelGroup(Alignment.LEADING)
-                .addComponent(sourceConfPanel.getForm(), GroupLayout.PREFERRED_SIZE, GroupLayout.PREFERRED_SIZE,
+                .addComponent(sourceConfPanel.getFormSourceConfig().getContainer(), GroupLayout.PREFERRED_SIZE, GroupLayout.PREFERRED_SIZE,
```

Comment:
This class must not know anything about how the internals of `SourceConfigurationPanel` work. I think it was OK like it was before.

All the things we need to do with the views must happen inside `SourceConfigurationPanel`.

So, let's revert this class and see wherever else we can work this out.

Reply:
Think about encapsulation:
- `PullTransferPanel` is a UI component that is composed of `SourceConfigurationPanel` and other children components
- `SourceConfigurationPanel` encapsulates the UI and behavior of the top panel of this UI, which lets the user select a source for Pulling/Pushing forms
- Does it make sense to let the `PullTransferPanel` the logic of changing `SourceConfigurationPanel`'s internal state and views? This would break encapsulation. Then, **we can't inject it with an `onSuccessCallback`**

Reply:
So let us maintain a `current form` variable in SourceConfigPanel. We will pass this to PullTransferPanel whenever it requests a top panel